lib/ansible/modules/cloud/amazon/redshift_security_group.py

```diff
@@ -0,0 +1,226 @@
+    try:
+        redshift.delete_cluster_security_group(group_name)
+    except boto.exception.JSONResponseError as e:
+        module.fail_json(msg=str(e))
+
+    return True
+
+
+def create_group(module, redshift):
+
+    group_name = module.params.get('group_name')
+    description = module.params.get('description')
+    rules = module.params.get('rules')
+
+    try:
+        group = []
+
+        sec_groups = redshift.describe_cluster_security_groups(group_name)
+        for sec_group in sec_groups['DescribeClusterSecurityGroupsResponse']['DescribeClusterSecurityGroupsResult']['ClusterSecurityGroups']:
+            if sec_group['ClusterSecurityGroupName'] == group_name:
+                group = sec_group
+                break
+
+        _clear_rules(module, redshift, group)
+    except boto.redshift.exceptions.ClusterSecurityGroupNotFound:
+
+        try:
+            redshift.create_cluster_security_group(group_name, description)
+        except boto.exception.JSONResponseError as e:
+            module.fail_json(msg=str(e))
+
+    # Process all rule properties.
+    if rules is not None:
+        if not isinstance(rules, list):
+            module.fail_json(msg='Rules needs to be a list of CIDR or EC2 security groups')
+
+    try:
+        for rule in rules:
+            group_kwargs = {}
+            if 'cidr' in rule:
+                group_kwargs['cidrip'] = rule['cidr']
+            elif 'sg_name' in rule:
+                group_kwargs['ec2_security_group_name'] = rule['sg_name']
+                if 'sg_owner' in rule:
+                    group_kwargs['ec2_security_group_owner_id'] = rule['sg_owner']
+
+            redshift.authorize_cluster_security_group_ingress(group_name, **group_kwargs)
+
+    except boto.exception.JSONResponseError as e:
+        module.fail_json(msg=str(e))
+
+    return True
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            state       = dict(choices=['present', 'absent'], required=True),
+            group_name  = dict(aliases=['name'], default='Default'),
+            description = dict(),
+            rules       = dict(type='list')
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec
+    )
+
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+
+    state = module.params.get('state')
+
+    region, url, aws_connect_params = get_aws_connection_info(module)
+
+    if not region:
+        module.fail_json(msg=str("Region not specified and unable to determinate region from EC2_REGION."))
+
+    # Connect to the Redshift endpoint.
+    try:
+        conn = connect_to_aws(boto.redshift, region, **aws_connect_params)
+    except boto.exception.JSONResponseError as e:
+        module.fail_json(msg=str(e))
+
+    if state == 'absent':
+        changed = delete_group(module, conn)
+    elif state == 'present':
+        changed = create_group(module, conn)
+
+    module.exit_json(changed=changed)
```

Comment:
@willthames The underlying function from boto doesn't return/expose any kind of id and also when you want to use the security group later you always use its name.
I could return the name, but I'm not sure if that's good practice since it's already defined in the playbook.